frontend/src/modules/settings/containers/DisplayTab.tsx

```diff
@@ -189,6 +243,12 @@ export const DisplayTab = ({ onSettingChange }: Props): JSX.Element => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [date, period, minute, hour, day, month, year, unit, theme, brightness]);
```

Comment:
What functionality does this `useEffect` enable?

Reply:
It listens for value change for all the states mentioned & triggers callback to parent component `SettingsPage` with updated values.